lib/cfg-grammar.y

```diff
@@ -1485,6 +1487,7 @@ source_reader_option
         /* NOTE: plugins need to set "last_reader_options" in order to incorporate this rule in their grammar */
 
 	: KW_CHECK_HOSTNAME '(' yesno ')'	{ last_reader_options->check_hostname = $3; }
+  | KW_CHECK_PROGRAM '(' yesno ')' { last_reader_options->check_program = $3; }
```

Comment:
The corresponding setting is missing from `log_reader_options_init()`.

(Setting `check-program(yes)` has no effect.)